### DISCUSSION TITLE
NextJS Bundle Splitting in Multi-Zone deployment

### DISCUSSION THREAD

**Discussion by hainessss** [vercel/next.js#23337] [Help]
Hello Everyone, I am curious how using a multi-zone (multiple applications stitched via http proxy) NextJS deployment works as far as the bundle splitting goes and how the browser handles them.  I'd imagine NextJS has to get all the bundles that the other app has built upon navigating to it? For example, Let's say there are two apps served at the routes: /users and /teams.  Both apps use the same react table component. When I navigate from /users to /teams will the code chunk for that table exist twice in the browser?  Or is next smart enough to not fetch it, if so how? If not, what are the performance implications of multi-zones? Will the bundle just grow and grow? 
Thanks in advance!

**Comment by hainessss** [comment]
Some has got to know? @timneutkens ?

**Reply by OZZlE** [reply]
On a somewhat related area, if you have multiple zones then react/nextjs/etc isn't included for each zone right? I hope zones is solution to the MFE problems of last years - every MFE including their own framework even when it's the same one as another?
